### ISSUE TITLE
TERRAFORM_PLAN_HAS_CHANGES doesn't work accross stages

### ISSUE THREAD

**Issue by philthynz** [charleszipp/azure-pipelines-tasks-terraform#232]
I have the following pipeline task in a separate stage:

```
    - task: TerraformCLI@0
      displayName: Terraform Plan
      inputs:
        command: 'plan'
        workingDirectory: '$(Build.SourcesDirectory)'
        environmentServiceName: '$(ServiceConnection)'
        commandOptions: '-out=$(System.DefaultWorkingDirectory)/terraform.tfplan -detailed-exitcode'
        allowTelemetryCollection: false
        publishPlanResults: true
```

And then a condition in a new stage:

```
- stage: terraform_apply
  displayName: Terraform Apply
  dependsOn: terraform_plan
  condition: and(succeeded(), eq(variables['TERRAFORM_PLAN_HAS_CHANGES'], 'true'))
```

This doesn't seem to work across stages. Can someone confirm?

**Comment by charleszipp** [comment]
@philthynz I believe this does not currently work because the `TERRAFORM_PLAN_HAS_CHANGES` is not set as an output variable by the TerraformCLI task. This was due originally to a limitation with the `azure-pipelines-task-lib` library. It did not expose an api to set output variables. Related #137 

If it were set as an output variable, then you should be able to share across stages using stage dependencies in the pipeline yaml. A workaround may be to use the bash task to emit the variable as an output via echo
```
stage: One
 - job: A
    steps:
     - bash: echo "##vso[task.setvariable variable=TF_OUT_TERRAFORM_PLAN_HAS_CHANGES;isOutput=true]$(TERRAFORM_PLAN_HAS_CHANGES)"
       name: out_tf_plan_has_changes
```

Then could reference from a subsequent stage as
```yaml
- stage: Two
  - job: B
    variables:
      # map the output variable from A into this job
      varFromA: $[ stageDependencies.One.A.outputs['out_tf_plan_has_changes.TF_OUT_TERRAFORM_PLAN_HAS_CHANGES'] ]
    steps:
    - script: echo $(TF_OUT_TERRAFORM_PLAN_HAS_CHANGES) # this step uses the mapped-in variable
```

**Comment by charleszipp** [comment]
Closing as this will be addressed with #137. Updated that issue to make sure this variable is included as part of that.
